namer/consul/src/main/scala/io/buoyant/namer/consul/SvcAddr.scala

```diff
@@ -76,40 +77,19 @@ private[consul] object SvcAddr {
             )
             stopped = true
             Future.Unit
-          case Throw(Failure(Some(err: ConnectionFailedException))) =>
-            log.warning(
-              "consul datacenter '%s' service '%s' retrying on connection" +
-                " failure: %s",
-              datacenter, key.name, err
-            )
-            // Drop the index, in case it's been reset by a consul restart
-            loop(None)
+
           case Throw(e) =>
-            // an error occurred. if we've previously seen a good state, fall
-            // back to that state; otherwise, fail.
+            // do not update state, log error and continue polling with backoff
             stats.errors.incr()
-            lastGood match {
-              case Some(addr) =>
-                // if we've already seen a good state, fall back to that and
-                // try again.
-                state() = addr
-                log.warning(
-                  "consul datacenter '%s' service '%s' observation error %s," +
-                    " falling back to last good state",
-                  datacenter, key.name, e
-                )
-                loop(index0)
-              case None =>
-                // if no previous good state was seen, treat the exception
-                // as effectively fatal to the service observation.
-                state() = Addr.Failed(e)
-                log.error(
-                  "consul datacenter '%s' service '%s' observation error %s," +
-                    " no previous good state to fall back to!",
-                  datacenter, key.name, e
-                )
-                Future.exception(e)
-            }
+            log.log(
+              failureLogLevel,
+              "consul datacenter '%s' service '%s' observation error %s." +
+                " Last known state is %s",
+              datacenter, key.name, e, lastGood
+            )
+            val backoff #:: nextBackoffs = backoffs
+            // subsequent errors are logged as DEBUG
+            Future.sleep(backoff).before(loop(None, nextBackoffs, Level.DEBUG))
```

Comment:
Passing the log level so that it's changed to debug on subsequent calls is very elegant, I like that 👍